embrace-android-features/src/main/kotlin/io/embrace/android/embracesdk/internal/capture/activity/UiLoadEventEmitter.kt

```diff
@@ -0,0 +1,149 @@
+package io.embrace.android.embracesdk.internal.capture.activity
+
+import android.app.Activity
+import android.app.Application.ActivityLifecycleCallbacks
+import android.os.Build
+import android.os.Bundle
+import io.embrace.android.embracesdk.internal.clock.nanosToMillis
+import io.embrace.android.embracesdk.internal.session.lifecycle.ActivityLifecycleListener
+import io.embrace.android.embracesdk.internal.utils.VersionChecker
+import io.opentelemetry.sdk.common.Clock
+
+/**
+ * Maps [ActivityLifecycleCallbacks] events to [UiLoadEvents] depending on the current state of the app and capabilities of the OS.
+ *
+ * The purpose of this is to leverage Activity lifecycle events to provide data for the underlying workflow to bring a new Activity on
+ * screen. Due to the varying capabilities of the APIs available on the different versions of Android, the precise triggering events for
+ * the start and intermediate steps may differ.
+ *
+ * See [UiLoadTraceEmitter] for details.
+ */
+class UiLoadEventEmitter(
+    private val uiLoadEvents: UiLoadEvents,
+    private val clock: Clock,
+    private val versionChecker: VersionChecker,
+) : ActivityLifecycleListener {
+
+    override fun onActivityPreCreated(activity: Activity, savedInstanceState: Bundle?) {
+        create(activity)
+    }
+
+    override fun onActivityCreated(activity: Activity, bundle: Bundle?) {
+        if (!versionChecker.firePrePostEvents()) {
+            create(activity)
+        }
+    }
+
+    override fun onActivityPostCreated(activity: Activity, savedInstanceState: Bundle?) {
+        createEnd(activity)
+    }
```

Comment:
Why is there a version check for activityCreated but not activityPostCreated? (This function and others are only available in API 29)

Reply:
So this is the reason why I did the changes further down the stack to separate the implementation between OS versions - this won't be called in pre-29 lifecycles, and having it called implies it's 29+. 

Yeah. Not great. Which is why this was refactored out in #1683 lol